code/datums/components/tackle.dm

```diff
@@ -80,6 +80,14 @@
 	if(!A || !(isturf(A) || isturf(A.loc)))
 		return
 
+	if(HAS_TRAIT(user, TRAIT_MOVE_FLOATING))
+		to_chat(user, span_warning("You can't tackle while floating!"))
+		return
```

Comment:
Wouldn't it be better to check if you're floating, and if you don't have anything to jump off of?

Reply:
That is how it worked on the fix+ moth PR but as that was one of the moth buffs I had to atomize it out.